### PR TITLE
add nuxt-vue3-google-signin (#429)

### LINKED ISSUE #429: Add `nuxt-vue3-google-signin`
## Features
`nuxt-vue3-google-signin` allows users to integrate google sign-in hassle-free. This module uses [`vue3-google-signin`](https://vue3-google-signin.syetalabs.io/) under the hood

## Repository
 https://github.com/syetalabs/nuxt-vue3-google-signin

## Icon
https://github.com/syetalabs/vue3-google-signin/blob/main/docs/public/logo.svg

## Definitions

```yaml
name: nuxt-vue3-google-signin
description: Empower your Nuxt app with Google Sign-in hassle-free
repo: syetalabs/nuxt-vue3-google-signin
npm: 'nuxt-vue3-google-signin'
github: https://github.com/syetalabs/nuxt-vue3-google-signin
website: https://vue3-google-signin.syetalabs.io/
learn_more: https://vue3-google-signin.syetalabs.io/
category: Libraries
type: community
maintainers:
  - name: Kasun Vithanage
    github: kasvith
compatibility:
  nuxt: ^3.0.0
  requires: {}
```

### PR DESCRIPTION
**Pull request by kasvith**
Closes #429

### CODE REVIEW

File: modules/nuxt-vue3-google-signin.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: nuxt-vue3-google-signin
+description: Empower your Nuxt app with Google Sign-in hassle-free
```
Comment (danielroe) [inline modules/nuxt-vue3-google-signin.yml:2]:
```suggestion
description: Empower your Nuxt app with Google Sign-In, hassle-free
```

**Comment by Atinux** [comment]
Looks good to me 😊 

I believe the next step would be to have a nice integration to keep the user session but not required now for it to be merged.

Waiting also for a small review from @pi0

**Review by Atinux — APPROVED** [review]

**Comment by kasvith** [comment]
@Atinux is it advisable to use `runtimeConfig` instead of `moduleConfig` for this plugin? I had a discussion with @danielroe in DC about this

**Comment by Atinux** [comment]
I am wondering since this is only used at this module is purely on the vue part to not leverage the [app.config](https://v3.nuxtjs.org/guide/features/app-config) directly?

**Comment by kasvith** [comment]
So then maybe i'll stick into `app.config` as per docs mentioned already

```ts
export default defineNuxtConfig({
  modules: [
    'nuxt-vue3-google-signin'
  ],
  googleSignIn: {
    clientId: 'CLIENT ID OBTAINED FROM GOOGLE API CONSOLE',
  }
})
```

**Comment by pi0** [comment]
Thanks for PR @kasvith. I will review impl shortly. Options of google-signin seems to be public but generally, I wouldn't advice using `app.config` at least today for storing authentication related setup. Different flows can potentially include some secrets (PS: This feature is only available in edge)

**Comment by kasvith** [comment]
Hi, @pi0 can you review it again? we have changed to use `app.config`

**Comment by kasvith** [comment]
Hi @pi0 any comments on the PR?

**Comment by kasvith** [comment]
Thanks @danielroe @Atinux @pi0

**Pull request merged**